File: rmi-chat/src/common/IChatCallback.java
package common;

import java.rmi.Remote;
import java.rmi.RemoteException;

public interface IChatCallback extends Remote {

	void onAccept(String[] names) throws RemoteException;
	
	void onDeny() throws RemoteException;
	
	void onAddUser(String name) throws RemoteException;
	
	void onRemoveUser(String name) throws RemoteException;
	
	void onReceive(String from, String message) throws RemoteException;
	
	void onExit() throws RemoteException;
}

File: WcfChat/Common/IChatCallback.cs
using System.ServiceModel;

namespace Common
{
	[ServiceContract]
	public interface IChatCallback
	{
		[OperationContract(IsOneWay = true)]
		void OnAccept(string[] names);

		[OperationContract(IsOneWay = true)]
		void OnDeny();

		[OperationContract(IsOneWay = true)]
		void OnAddUser(string name);

		[OperationContract(IsOneWay = true)]
		void OnRemoveUser(string name);

		[OperationContract(IsOneWay = true)]
		void OnReceive(string from, string message);

		[OperationContract(IsOneWay = true)]
		void OnExit();
	}
}
